### PR TITLE
Add Changelog Build Step

### PR DESCRIPTION
**Pull request by micbar**
# Description
- Add an automated drone step to check if the build of the changelog is successful
- Add an automated drone step to commit the newly generated changelog to master after a PR merge.

## Motivation and Context
This would automate the whole changelog generation

## How Has This Been Tested?
- Manually

### New pipeline
Screenshot_2019-12-11 Build #21928 3 1 - owncloud core - Drone CI

### Generate
Screenshot_2019-12-11 Build #21928 3 2 - owncloud core - Drone CI

### Diff
Screenshot_2019-12-11 Build #21928 3 3 - owncloud core - Drone CI

### output
Screenshot_2019-12-11 Build #21928 3 4 - owncloud core - Drone CI

## After merge to master
- A new git commit will be created comitted to master
- CI will be skipped for that

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: changelog/README.md [hunk 1]
```diff
@@ -5,10 +5,15 @@ changelog before we are tagging a new release.
 
 ## Create Changelog items
 Create a file according to the [template](TEMPLATE) for each 
-changelog in the [unreleased](./unreleased) folder. The following change types are possible: `Bugfix, Change, Enhancement, Security`.
+changelog in the [Unreleased](./Unreleased) folder. The following change types are possible: `Bugfix, Change, Enhancement, Security`.
```
Comment (phil-davis) [inline changelog/README.md]:
```suggestion
changelog in the [unreleased](./unreleased) folder. The following change types are possible: `Bugfix, Change, Enhancement, Security`.
```
The folder in core is currently called `unreleased` (lowercase).
Did you intend to change it?

File: changelog/README.md [hunk 2]
```diff
@@ -5,10 +5,15 @@ changelog before we are tagging a new release.
...
 
-## Generate the Changelog 
-- copy the files from the [unreleased](./unreleased) folder into a folder matching the
-schema `10.0.0_2019-09-01` to tag and 
+## Automated Changelog build and commit
+- After each merge to master, the CHANGELOG.md file is automatically updated and the new version will be committed to the master while skipping the CI.
```
Comment (phil-davis) [inline changelog/README.md]:
```suggestion
- After each merge to master, the CHANGELOG.md file is automatically updated and the new version will be committed to master while skipping CI.
```

A couple of "the" are not needed.

File: changelog/README.md [hunk 3]
```diff
@@ -5,10 +5,15 @@ changelog before we are tagging a new release.
...
+
+## Create a new Release
+- copy the files from the [Unreleased](./Unreleased) folder into a folder matching the
```
Comment (phil-davis) [inline changelog/README.md]:
```suggestion
- copy the files from the [nreleased](./unreleased) folder into a folder matching the
```

**Review by phil-davis — COMMENTED** [review]
Just small comments.

What will happen on a PR that is test-only and does not have/need a changelog entry?

The regenerated CHANGELOG.md file will be unchanged, so the extra commit will be empty. I guess we will see if something explodes or...

File: changelog/README.md (hunk 3 above)
Comment (micbar) [reply changelog/README.md]: missing "u"

Comment (phil-davis) [reply changelog/README.md]: Fixed my comment

**Review by phil-davis — APPROVED** [review]
LGTM

**Comment by micbar** [comment]
@phil-davis If the diff is empty, git will not create a new commit.

**Pull request merged**